lib/styles.js

```diff
@@ -340,4 +342,133 @@ module.exports = function (expect) {
             this.sp().commentBlock(pen);
         }
     });
+
+    expect.addStyle('diffFragment', function (text, baseStyle, specialCharStyle, markUpSpecialCharacters) {
+        if (markUpSpecialCharacters) {
+            text.split(specialCharRegExp).forEach(function (part) {
+                if (specialCharRegExp.test(part)) {
+                    this[specialCharStyle || baseStyle](utils.escapeChar(part));
+                } else {
+                    this[baseStyle](part);
+                }
+            }, this);
+        } else {
+            this[baseStyle](text);
+        }
+    });
+
+    expect.addStyle('stringDiff', function (actual, expected, options) {
+        options = options || {};
+        var type = options.type || 'WordsWithSpace';
+        var diffLines = [];
+        var lastPart;
+        stringDiff.diffLines(actual, expected).forEach(function (part) {
```

Comment:
That's also what I arrived at. I attempted it once, but got stuck when I ran into the "internal" diff that the handling of `replace`d chunks does.